### PR TITLE
Move swtpm_localca sources from samples/ to src/swtpm_localca/

### PR DESCRIPTION
**Pull request by nchevsky**
- Relocates `swtpm_localca`'s sources out of `samples/` (where they no longer belong now that `swtpm_localca` is a binary) to `src/swtpm_localca/`, joining the rest of the compiled programs.
- Tests now call the `swtpm_localca` binary directly at the location where it was built, as all tests do with all the other compiled programs.
- Simplifies `samples/swtpm-localca.in` by removing the now-unnecessary logic to selectively invoke `swtpm_localca` from different locations (`samples/` when running tests vs. `/usr/bin/` post-installation).
- Reverts the now-unnecessary `samples/swtpm-localca.2inst` hacks introduced [here](https://github.com/stefanberger/swtpm/commit/50f7d49ad3aa92117a9f4701f315c6d567aa21bb).

### CODE REVIEW

File: samples/Makefile.am [hunk 1]
```diff
@@ -9,57 +9,24 @@ samplessysconfdir = $(sysconfdir)
...
 	swtpm-localca.options \
 	swtpm_setup.conf
 
-noinst_HEADERS = \
-	swtpm_localca.h \
-	swtpm_localca_utils.h
-
-bin_PROGRAMS = \
-	swtpm_localca
-
```
Comment (nchevsky) [inline samples/Makefile.am:25]: `swtpm_localca` is now built by its own makefile in `src/swtpm_localca/`, like all the other compiled programs.

File: samples/Makefile.am [hunk 2]
```diff
@@ -9,57 +9,24 @@ samplessysconfdir = $(sysconfdir)
 
 samplesconf_SCRIPTS = \
 	swtpm-create-tpmca \
-	swtpm-create-user-config-files
+	swtpm-create-user-config-files \
+	swtpm-localca
```
Comment (nchevsky) [inline samples/Makefile.am:13]: Reverted to the state prior to the `samples/swtpm-localca.2inst` hacks introduced [here](https://github.com/stefanberger/swtpm/commit/50f7d49ad3aa92117a9f4701f315c6d567aa21bb).

File: samples/Makefile.am [hunk 3]
```diff
@@ -9,57 +9,24 @@ samplessysconfdir = $(sysconfdir)
...
 		chmod 0750 $(DESTDIR)$(localstatedir)/lib/swtpm-localca || true; \
 	fi
 
-install-exec-local:
-	$(MKDIR_P) $(DESTDIR)$(samplesconfdir)
-	$(INSTALL_SCRIPT) swtpm-localca.2inst $(DESTDIR)$(samplesconfdir)/swtpm-localca
-
-uninstall-local:
-	rm -f $(DESTDIR)$(samplesconfdir)/swtpm-localca
-
```
Comment (nchevsky) [inline samples/Makefile.am:62]: Reverted to the state prior to the `samples/swtpm-localca.2inst` hacks introduced [here](https://github.com/stefanberger/swtpm/commit/50f7d49ad3aa92117a9f4701f315c6d567aa21bb).

File: samples/swtpm-localca.in [hunk 4]
```diff
@@ -1,10 +1,5 @@
 #!/usr/bin/env sh
 
-swtpm_localca_exe="$(dirname "$0")/swtpm_localca"
-if ! [ -x "$swtpm_localca_exe" ]; then
-  swtpm_localca_exe=@BINDIR@/swtpm_localca
-fi
-
-$swtpm_localca_exe "$@"
```
Comment (nchevsky) [inline samples/swtpm-localca.in:8]: This logic is unnecessary now that tests call `swtpm_localca` directly at the location where it was built, as all tests do with all the other compiled programs. The wrapper script is now only used post-installation, and no longer by tests.

File: src/swtpm_localca/Makefile.am [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	$(top_builddir)/src/utils/libswtpm_utils.la
+
+swtpm_localca_LDFLAGS = \
+	-L$(top_builddir)/src/utils -lswtpm_utils \
+	$(GLIB_LIBS) \
+	$(HARDENING_LDFLAGS)
+
+swtpm_localca_SOURCES = \
+	swtpm_localca.c \
+	swtpm_localca_utils.c
```
Comment (nchevsky) [inline src/swtpm_localca/Makefile.am:32]: All this was brought over unmodified from `samples/Makefile.am`.

**Comment by stefanberger** [comment]
Thanks. Can you copy the text at the top into the commit text ?

**Pull request merged**